core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -73,21 +79,26 @@ public Optional<Result> get(Get get) throws CrudException {
     List<String> originalProjections = new ArrayList<>(get.getProjections());
     Snapshot.Key key = new Snapshot.Key(get);
     readUnread(key, get);
-    return createGetResult(key, originalProjections);
+    return createGetResult(key, get, originalProjections);
   }
 
   @VisibleForTesting
   void readUnread(Snapshot.Key key, Get get) throws CrudException {
-    if (!snapshot.containsKeyInReadSet(key)) {
+    if (!snapshot.contains(get)) {
       read(key, get);
     }
   }
 
   private void read(Snapshot.Key key, Get get) throws CrudException {
     Optional<TransactionResult> result = getFromStorage(get);
     if (!result.isPresent() || result.get().isCommitted()) {
-      snapshot.put(key, result); // for read set
-      snapshot.put(get, key); // for extra-read using Get
+      if (result.isPresent() || get.getConjunctions().isEmpty()) {
```

Comment:
It's correct since this comment explains using contraposition, i.e., the condition and `else` part in `if (...) { } else { do nothing }`. But, definitely, it's difficult to understand. Is the following better?

> Keep the read set latest to create the before image using it when the transaction writes or deletes this record. However, we update it only if a get operation has no conjunction or the result exists. This is because we don’t know whether the record actually does not exist due to the conjunction.

Reply:
Thanks! The revised version looks far easier to understand to me. Especially the sentence `This is because` is very helpful for me.

Reply:
Fixed in 4463ef8.

Reply:
`it` means the latest record, and `using` modifies `create`. Does the example in my reply to your comment for scan help to understand?

> Can we say like this?
> `This is because we don’t know whether the record actually exists or not due to the conjunction.`

Yes, it's far better. Thanks.

Reply:
Fixed in 67446f1 once, but I would like to further improve it if you have additional questions.